src/angular-bootstrap-toggle.js

```diff
@@ -1,289 +1,313 @@
-                    // If set, we will use their values for onClass and offClass respectively
-                    if (self.onstyle) {
-                      self.onClass = self.onstyle;
-                    }
-                    if (self.offstyle) {
-                      self.offClass = self.offstyle;
-                    }
-                    // Special treatment for style, now deprecated and replaced with toggleClass
-                    if (self.style) {
-                      self.toggleClass = self.style;
-                    }
+						var labels = self.element.find('label');
+						var spans = self.element.find('span');
+						var divs = self.element.find('div');
 
-                    // Special case: empty on and off labels (replace with blank space)
-                    if (self.on === '') {
-                      self.on = '&nbsp;';
-                    }
-                    if (self.off === '') {
-                      self.off = '&nbsp;';
-                    }
+						self.wrapperElement = divs[0];
+						self.onElement = labels[0];
+						self.offElement = labels[1];
+						self.handleElement = spans[0];
 
-                    this.init = function (ngModelCtrl_) {
-                        ngModelCtrl = ngModelCtrl_;
+						var onElement = angular.element(self.onElement);
 
-                        var labels = self.element.find('label');
-                        var spans  = self.element.find('span');
-                        var divs   = self.element.find('div');
+						// Set wigget's visible text such as On/Off or Enable/Disable
+						onElement.html(self.on);
+						angular.element(self.offElement).html(self.off);
 
-                        self.wrapperElement = divs[0];
-                        self.onElement = labels[0];
-                        self.offElement = labels[1];
-                        self.handleElement = spans[0];
+						// Set the button size
+						angular.element(self.wrapperElement).addClass(self.size);
+						onElement.addClass(self.size);
+						angular.element(self.offElement).addClass(self.size);
+						angular.element(self.handleElement).addClass(self.size);
 
-                        // Set wigget's visible text such as On/Off or Enable/Disable
-                        angular.element(self.onElement).html(self.on);
-                        angular.element(self.offElement).html(self.off);
+						// Set the toggleClass on the wrapper
+						angular.element(self.wrapperElement).addClass(self.toggleClass);
 
-                        // Set the button size
-                        angular.element(self.wrapperElement).addClass(self.size);
-                        angular.element(self.onElement).addClass(self.size);
-                        angular.element(self.offElement).addClass(self.size);
-                        angular.element(self.handleElement).addClass(self.size);
+						if (!onElement.is(':visible')) {
```

Comment:
Last thoughts about this change: in an AngularJS application, visibility is often an issue with directives that deal with their appearance, size, etc. That's why it is recommended to use AngularJS directives to handle things like collapse, tabs, etc.

In the case of collapse, shouldn't you be using `uib-collapse` instead of pure CSS? see [UI Bootstrap](https://angular-ui.github.io/bootstrap/#!#collapse)

I mean: I totally understand that this change makes things easier on the developer, but aren't we shoehorning/arm-twisting `<angular-bootstrap-toggle>` to support a use case that shouldn't happen if the application was using dedicated AngularJS components for collapse (and others) rather than "hacking" into CSS directly?

Happy to discuss this! :-)

Reply:
No Application can be working without some parts of pure css styling and in the end this is what bootstrap is for. Why should you write/use a component within angularjs if the same function is available in pure css that the browser supports directly?.

This bug is not even a bit related to angular itself but to every application case where the toggle is not "visible" but exists in DOM. This is a simple bug within the directive that expects that the control is always visible in DOM. As within my bug ticket i have created an example just with some plain CSS. FMPOV if you are relying on calulation of your own size, you MUST ensure that this is happen when there is any meaningful result to be expected. Every modern application, let it be Angularjs, Knockout, Aurellia ect _can_ hide html in dom and you can bet that some devloper will do this if they are able to.

Reply:
I totally agree with you @JPVenson. I'm really just questioning the philosophy here.
Anyway, apart from the quote changes (see my previous comment), I'm fine with your change and the principle (and you made sure the Observer was running strictly only when necessary, which is good).
I'll approve once you got the double quotes back when appropriate in the comment sections ;-) (yeah... it's a pain, I'm sorry).
But I won't be able to merge. Only @ziscloud can!